storage/types.go

```diff
@@ -277,6 +282,34 @@ func (n *NodeID) String() string {
 	return r.String()
 }
 
+// AsKey returns a string identifier for this NodeID suitable for
+// short term use e.g. as a Map key. It is more efficient to use this than
+// String() as it's not constrained to return a binary string.
+func (n *NodeID) AsKey() string {
+	var b strings.Builder
+	fullBytes := n.PrefixLenBits / 8
+	bitsLeft := n.PrefixLenBits % 8
+
+	// Note: all Builder write methods are documented never to return errors.
+	// Write the length first.
+	b.WriteString(strconv.Itoa(n.PrefixLenBits))
+	b.WriteRune(':')
+	// We can do all the full bytes in one go.
+	if fullBytes > 0 {
+		buf := make([]byte, hex.EncodedLen(fullBytes))
+		hex.Encode(buf, n.Path[:fullBytes])
+		b.Write(buf)
+	}
+	// If there's bits left over write them out.
+	if bitsLeft > 0 {
+		bits := n.Path[fullBytes] & leftmask[bitsLeft]
+		b.WriteByte(hexChars[bits>>4])
```

Comment:
```fmt.Fprintf(&b, "%x", bits)```?

Reply:
If you mean Sprintf that's more string copying.

Reply:
`Fprintf` writes directly to the passed in output buffer.

Reply:
Tried this locally, `Fprintf` here makes the thing 25% slower (prob. due to indirection).